packages/filebrowser-extension/src/index.ts

```diff
@@ -394,7 +376,10 @@ const browserWidget: JupyterFrontEndPlugin<void> = {
     settings: ISettingRegistry,
     toolbarRegistry: IToolbarWidgetRegistry,
     translator: ITranslator,
-    labShell: ILabShell
+    labShell: ILabShell,
+    // Wait until file browser commands are ready before activating file browser widget
+    fileBrowserCommands: null,
```

Comment:
@brichet why is this implicit dependency needed?

Maybe there is a way to reorganize the content of the plugins to not depend on the plugin action order at runtime?

Reply:
The `IFileBrowserCommands` was already required before this PR, only the `ICommandPalette` wasn't.
But I need to 'consume' this parameter to get the command palette.

The requirement of `IFileBrowserCommands` come from https://github.com/jupyterlab/jupyterlab/pull/12369, so maybe @fcollonval and @afshin have a better idea on this. I tried without this requirement and everything seems to work well at first sight.

Reply:
> The requirement of `IFileBrowserCommands` come from #12369

Ah thanks for linking this :+1: 

Since there was a new line and a comment in the diff I thought it was added in the current PR. So we can consider having another look at it separately. The top comment of https://github.com/jupyterlab/jupyterlab/pull/12369 mentions some race condition and maybe there is another way to solve it.